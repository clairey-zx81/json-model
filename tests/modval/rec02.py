# Generated for model: ./rec02.model.json
#
# Generated by jmc (JSON Model Compiler) (https://github.com/clairey-zx81/json-model)
#

from typing import Callable
import re as re
import datetime
import urllib.parse

type Jsonable = None|bool|int|float|str|list[Jsonable]|dict[str, Jsonable]
type CheckFun = Callable[[Jsonable, str], bool]
type PropMap = dict[str, CheckFun]
type TagMap = dict[None|bool|float|int|str, CheckFun]

json_model_0_may: PropMap

# define "$foo" ($.foo)
def json_model_1(value: Jsonable, path: str) -> bool:
    # $.foo
    result = json_model_0(value, path)
    return result


# object $
def json_model_0(value: Jsonable, path: str) -> bool:
    if not isinstance(value, dict):
        return False
    for prop, model in value.items():
        assert isinstance(prop, str)
        if prop in json_model_0_may:  # may
            if not json_model_0_may[prop](model, f"{path}.{prop}"):
                return False
        else:  # no catch all
            return False
    return True



# entry function check_model
def check_model(value: Jsonable, path: str = "$") -> bool:
    return json_model_0(value, path)


# object properties maps
json_model_0_may = {
    "foo": json_model_1,
}


