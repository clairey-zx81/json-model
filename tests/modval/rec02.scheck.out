rec02.0.true.json: PASS
rec02.1.true.json: PASS
rec02.2.true.json: PASS
rec02.3.false.json: FAIL
rec02.4.false.json: FAIL
